components/WeeklyOverview/index.js

```diff
@@ -1,25 +1,84 @@
-import React from 'react';
+import React, { useState } from 'react';
 import styled from 'styled-components';
 import { dummyMedications } from '../../dummydata';
 
 export default function WeeklyOverview() {
   const weekdays = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'];
-  const supplementsByWeekday = {};
 
-  weekdays.forEach((weekday) => {
-    supplementsByWeekday[weekday] = dummyMedications
-      .filter((supplement) => supplement.weekday === weekday)
-      .slice(0, 2);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [selectedMedicationsByWeekday, setSelectedMedicationsByWeekday] = useState(() => {
+    const selectedMeds = {};
+    weekdays.forEach((weekday) => {
+      selectedMeds[weekday] = dummyMedications[0];
+    });
+    return selectedMeds;
   });
+  const [supplementsByWeekday, setSupplementsByWeekday] = useState(() => {
+    const supplements = {};
+    weekdays.forEach((weekday) => {
+      supplements[weekday] = [];
+    });
+    return supplements;
+  });
+
+  const handleAddSupplement = (weekday) => {
+    setSupplementsByWeekday((prevSupplements) => ({
+      ...prevSupplements,
+      [weekday]: [...prevSupplements[weekday], selectedMedicationsByWeekday[weekday]],
+    }));
+  };
+
+  const handleSearch = () => {
+    const filteredData = {};
```

Comment:
Maybe you could find a more descriptive word than "data", maybe "filteredSearchResults" or sth. like that? :)